test/README.md

```diff
@@ -1,95 +1,296 @@
+"numberLoggerImage": "localhost:5000/microsoft/number-logger:latest-linux-amd64",
+"genericMqttTesterImage": "localhost:5000/microsoft/generic-mqtt-tester:latest-linux-amd64",
+"diagnosticsImage": "localhost:5000/microsoft/diagnostics:latest-linux-amd64",
+"metricsValidatorImage": "localhost:5000/microsoft/metrics-validator:latest-linux-amd64",
+"caCertScriptPath": "/home/azureuser/iotedge/test/certs",
+"dpsIdScope": "0ne003A5268",
+"rootCaCertificatePath": "/home/azureuser/iotedge/test/certs/certs/azure-iot-test-only.root.ca.cert.pem",
+"rootCaPrivateKeyPath": "/home/azureuser/iotedge/test/certs/private/azure-iot-test-only.root.ca.key.pem",
+"verbose": "true",
+"logFile": "/home/azureuser/iotedge/logs/logfile",
+"registries":
+        [
+                {
+                        "address": "localhost:5000",
+                        "username": "username"
+                }
+        ]
+}
+~~~
+
+### Test secrets
+
+The tests also expect to find several _secret_ parameters. While these can technically be added to `context.json`, it is recommended that you create environment variables and make them available to the test framework in a way that avoids committing them to your shell's command history or saving them in clear text on your filesystem. When set as environment variables, all secret parameters must be prefixed with `E2E_`. Parameter names are case-**in**sensitive; they're only shown in uppercase here to follow the common convention for environment variables, and to stand out as secrets.
+
+| Name | Required | Description |
+|------|----------|-------------|
+| `E2E_DPS_GROUP_KEY` | * | The symmetric key of the DPS [enrollment group](https://docs.microsoft.com/en-us/azure/iot-dps/concepts-service#enrollment-group) to use. Used when running any DPS tests, ignored otherwise. |
+| `E2E_IOT_HUB_CONNECTION_STRING` | ✔ | Hub-scoped IoT Hub connection string that can be used to get/add/remove devices, deploy edge configurations, and get/update module twins. |
+| `E2E_EVENT_HUB_ENDPOINT` | ✔ | Connection string used to connect to the Event Hub-compatible endpoint of your IoT Hub, to listen for D2C events sent by devices or modules. |
+| `E2E_REGISTRIES__{n}__PASSWORD` || Password associated with a container registry entry in the `registries` array of `context.json`. `{n}` is the number corresponding to the (zero-based) array entry. For example, if you specified a single container registry in the `registries` array, the corresponding parameter would be `[E2E_]REGISTRIES__0__PASSWORD`. |
+| `E2E_ROOT_CA_PASSWORD` || The password associated with the root certificate specified in `rootCaCertificatePath`. |
+| `E2E_BLOB_STORE_SAS` || The sas token used to upload module logs and support bundle in the tests. |
+
+_Note: the definitive source for information about test parameters is `test/Microsoft.Azure.Devices.Edge.Test/helpers/Context.cs`._
+
+#### Sample environnment settings
+~~~ sh
+export E2E_IOT_HUB_CONNECTION_STRING="HostName=..."

```

Comment:
Earlier it says

> it is recommended that you create environment variables and make them available to the test framework in a way that avoids committing them to your shell's command history...

But then here, in our example script, we're creating environment variables in a way that commits them to our shell's command history (same for setting up the password with htpasswd earlier in the doc).

I usually use `read -s {var name}` to enter secrets on the command line, or `secret=$(az kv secret show ...)` if the values are in Key Vault. Both approaches keep secrets out of the command history. But I'm not sure the best way to do it here...

Reply:
Good call out - our example should be `secure by default`. The best way is to probably get it setup in a secret store, but I am instead going to leave it up to the developer to do as they see fit. 

My go to mechanism, more out of convenience, is to create a script and run that in my setup. While this approach keeps the secrets out of the shell's command history, it does have the drawback of storing these secrets in the filesystem

Reply:
I agree our example should be secure by default. How best to balance that with simplicity, ease of use, etc. is your call...